redis/src/main/scala/zio/redis/api/SortedSets.scala

```diff
@@ -2,24 +2,25 @@ package zio.redis.api
 
 import zio.redis.Input._
 import zio.redis.Output._
-import zio.redis.RedisCommand
+import zio.redis.{ Output, RedisCommand }
 
 trait SortedSets {
   final val bzPopMax = RedisCommand("BZPOPMAX", Tuple2(DurationSecondsInput, NonEmptyList(StringInput)), ChunkOutput)
   final val bzPopMin = RedisCommand("BZPOPMIN", Tuple2(DurationSecondsInput, NonEmptyList(StringInput)), ChunkOutput)
 
   // TODO: represent two possible outputs as per the documentation
-  final val zAdd = RedisCommand(
-    "ZADD",
-    Tuple5(
-      StringInput,
-      OptionalInput(UpdateInput),
-      OptionalInput(ChangedInput),
-      OptionalInput(IncrementInput),
-      NonEmptyList(MemberScoreInput)
-    ),
-    LongOutput
-  )
+  final def zAdd[T](output: Output[T] = LongOutput) =
```

Comment:
Please start from `A`. Also, passed like this, any output (even invalid one) will suffice.

Reply:
indeed, hmm I need to think about some restriction only to String or Long then

Reply:
@mijicd do you have maybe some idea, vision about this ?

Reply:
ah well now I look at this another time, and dont like this idea ;)
Because each time user will need to be sure about redis output.

zAddWithIncr would be clear enough, about the output type at least.
One method -> one return type

wdyt ?

Reply:
We shouldn't propagate this to "user land" as it is already determined by presence or absence of increment input. Therefore, let's remove it from here, and determine it during parsing. I think @aleksandarskrbic did something similar in his latest iteration on hashes / geo test.

Reply:
We have a race condition in this conversation :). I'm fine with that approach, but let's see if @aleksandarskrbic and @anovakovic01 can recall the similar situation we had and the solution for it :).

Reply:
ok cool :)

Reply:
@maciejbak85 I've stumbled on a similar problem while working on Hash API, specifically HMGET command since we want to return a Map[Field, Value]. Problem is that Redis will return just a list of values. We decided to somehow wire together input and output so while parsing the response, we have information about a command and we can extract fields that users passed as arguments. This was an idea, I haven’t started implementation yet.

Reply:
I usually had situations where I could place everything under a single output type. In this case, you could create sum type which will represent an integer or double output. So something like this:
```scala
sealed trait ZAddReply
object ZAddReply {
  case class Number(value: Long) extends ZAddReply
  case class Score(value: Double) extends ZAddReply
}
```
and then just create `case object ZAddOutput extends Output[ZAddReply]`. In this case, `zAdd` will have `OptionalOutput(ZAddOutput)` as output parameter. WDYT? @maciejbak85 @mijicd

Reply:
@anovakovic01 minus I see is that user needs to be aware of what type redis returns, library user should not bother about it

Reply:
Well, yes and no, we can parse the output in sort of alternative / orElse fashion. Given possible terms, try them out. Still, it might be cumbersome to propagate.